include/QueryGenerator/QueryGenerator.php

```diff
@@ -1535,6 +1547,11 @@ private function isDateType($type) {
 		return ($type == 'date' || $type == 'datetime');
 	}
 
+	private function requiresSoundex($operator) {
+		$requiresSX = array('sx');
+		return in_array($operator, $requiresSX);
```

Comment:
just one line, directly return the result: `return in_array($operator, array('sx'));` there is no need to save the array in memory just to throw it away